docs/DesignDocs/LoggingFormat.md

```diff
@@ -0,0 +1,79 @@
+Logging Format
+=====
+
+When using the OE SDK and in trying to make use of the logs (e.g. via uploading them to Azure Log Analytics) it is nice to be able to parse and display all of the available logs including the ones generated by OE.
+
+Motivation
+----------
+
+Users should be able to specify and control their own logging format so that they can further parse the log files as they see fit.
+
+User Experience
+---------------
+
+- The user can set the `OE_LOG_DEVICE` environment variable to specify the log file location
+
+- The user can set the `OE_LOG_FORMAT` environment variable to specify the log format. The user will be given the below information:
+
+        - time in string format
+        - microseconds
+        - is enclave (E|H)
+        - log level
+        - thread id
+        - log message
+        - file log initiated from
+        - function log initiated from
+        - file line number log initiated from
+
+        so for example a log format could be set like this:
+
+        `OE_LOG_FORMAT=$'{\"e_ts\":\"%s.%06ldZ\",\"level\":\"(%s)%s\",\"tid\":\"tid(0x%lx)\",\"msg\":\"%s\",\"file\":\"%s\",\"func\":\"%s\",\"number":\"%s\"}\n'`
```

Comment:
I have some questions:
- What would be another good example where the use can specify log format other than JSON? Adding a second non-JSON example would help users understand how to leverage this feature.
- Can the log format skip unnecessary fields? Say I don't want the thread-id since my application is single-threaded.
- Can the user specify fields in different order via the format?
- Not sure if it is a typo: The format above specified `e_ts`, `level`, `tid` ... but the output below has fields in different order.

Reply:
Agree that a second non-JSON example would be helpful, but could be done in a future PR in my opinion.   If you want another standard format to use in examples, use syslog format.  There's examples in RFC 5424 and in various sources like the ones linked to as references from the wikipedia entry for syslog.

Reply:
@anakrish 

- We don't have another example right now we were only interested in the JSON use case at this point
- This PR does not offer the ability to skip unnecessary fields
- This PR does not offer the ability to define a different order
-  This was a result of pretty printing JSON. Re-ordered to avoid confusion

Reply:
@olgavrou Thanks for the answers.
@dthaler Thanks for the pointer to syslog.